src/main/java/gift/service/KakaoAuthService.java

```diff
@@ -0,0 +1,57 @@
+package gift.service;
+
+import gift.dto.KakaoMemberResponseDto;
+import gift.dto.KakaoTokenResponseDto;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClient;
+
+@Service
+public class KakaoAuthService {
+
+    @Value("${kakao.rest-api-key}")
+    private String kakaoRestApiKey;
+
+    @Value("${kakao.redirect-url}")
+    private String redirectUrl;
+
+    private final RestClient restClient = RestClient.create();
```

Comment:
```suggestion
        private final RestClient restClient = RestClient.builder()
            .baseUrl("https://kauth.kakao.com")
            .build();
```
클라이언트를 선언할 때 공통되는 부분들은 이렇게 미리 설정하는 방식을 많이 활용합니다!
지금은 그럴 필요가 없지만 RestClient 활용법에 대해 미리 알아두시기만 하면 좋을 것 같아요!
https://docs.spring.io/spring-framework/reference/integration/rest-clients.html